### PR TITLE
chore: [#187702425] Sync migrations with generators

### PR DESCRIPTION
**Pull request by SirSamuelJoseph**
## Description

Updates the generators and migration files to synchronize the fields present in each

### Ticket

This pull request resolves [#187702425](https://www.pivotaltracker.com/story/show/187702425).

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation on the engineering documentation website
- [x] I have not used any relative imports
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] I have checked for and removed instances of unused config from CMS
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [x] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: api/src/db/migrations/migrations.ts [hunk 1]
```diff
@@ -284,4 +285,7 @@ export const Migrations: MigrationFunction[] = [
   migrate_v137_to_v138,
   migrate_v138_to_v139,
   migrate_v139_to_v140,
+  migrate_v140_to_v141,
 ];
+
+export { generatev141UserData as CURRENT_GENERATOR } from "@db/migrations/v141_sync_migrated_and_newly_created_users";
```
Comment (SirSamuelJoseph) [inline api/src/db/migrations/migrations.ts:291]: Updating the documentation around this: when creating a new migration, in addition to adding it to the Migrations const, you should also update what is getting exported as the `CURRENT_GENERATOR`.

File: api/src/db/migrations/v141_sync_migrated_and_newly_created_users.ts [hunk 2]
```diff
@@ -0,0 +1,829 @@
+import { v140Business, v140UserData } from "@db/migrations/v140_modify_home_health_industry";
+import { randomInt } from "@shared/intHelpers";
+
+export const migrate_v140_to_v141 = (v140Data: v140UserData): v141UserData => {
```
Comment (SirSamuelJoseph) [inline api/src/db/migrations/v141_sync_migrated_and_newly_created_users.ts:4]: These migration specific updates are important to check in this PR to see if there are any errors in how we are bringing users up to current version of UserData.

**Review by aadedejifearless — COMMENTED** [review]
There are an issue with the Unit Test.

File: api/src/db/migrations/migrations.test.ts [hunk 3]
```diff
@@ -86,5 +134,29 @@ describe("migrations", () => {
...
+
+    it("runs all migrations and gets user to current version", () => {
+      let user: unknown = generateV0UserData({});
+      for (const func of Migrations) {
+        user = func(user);
+      }
+      // @ts-expect-error getting user version from unknown
+      expect(user.version).toEqual(CURRENT_VERSION);
+      expect(Migrations.length).toEqual(CURRENT_VERSION);
+    });
```
Comment (jphechter) [inline api/src/db/migrations/migrations.test.ts:146]: This is a great add and part of me is surprised we didn't have this before.

File: api/src/db/migrations/migrations.ts (hunk 1 above)
Comment (jphechter) [reply api/src/db/migrations/migrations.ts:291]: Could we add a test that checks if `CURRENT_GENERATOR` returns a user with the current version?

**Review by jphechter — APPROVED** [review]

File: api/src/db/migrations/migrations.ts (hunk 1 above)
Comment (seidior) [reply api/src/db/migrations/migrations.ts:291]: Do we have a doc on how to do migrations yet?

File: shared/src/userData.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export interface Business {
   readonly formationData: FormationData;
 }
 
-export const CURRENT_VERSION = 140;
+export const CURRENT_VERSION = 141;
```
Comment (alexander-littleton) [inline shared/src/userData.ts:32]: [sand]: Do we have a good enough reason to not just have this equal `Migrations.length`? I haven't been able to come up with one but curious on the thoughts of people who are more familiar with this code (everybody else)

**Pull request merged**